sklearn/utils/testing.py

```diff
@@ -372,14 +375,30 @@ def __exit__(self, *exc_info):
     assert_greater = _assert_greater
 
 
+def _assert_sparse_allclose(actual, desired, rtol=1e-7, atol=0, err_msg='',
+                            verbose=True):
+    """Check if two sparse arrays are equal up to the desired tolerance.
+
+    It compares the difference between `actual` and `desired` to
```

Comment:
The test compares the *absolute* value of the difference to the thing you mention. Not the difference. For more information take a look at the [documentation of `np.allclose`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html).